pokemon-league-participation-manager/src/main/java/cz/muni/fi/pa165/pokemon/league/participation/manager/entities/PokemonSpecies.java

```diff
@@ -0,0 +1,99 @@
+package cz.muni.fi.pa165.pokemon.league.participation.manager.entities;
+
+import cz.muni.fi.pa165.pokemon.league.participation.manager.enums.PokemonType;
+import java.util.Objects;
+
+/**
+ * Entity class for object PokemonSpecies.
+ * 
+ * @author Jiří Medveď 38451
+ */
+public class PokemonSpecies {

```

Comment:
General note on this class:
I'd make it (almost) immutable - Species doesn't hold any information which might change dynamically - they don't change names, types and pre-evolutions. The only thing we might want to change is ID (even that only in "set to not null once and don't change afterwards" mechanic, really), but then I'm not sure there is a use case for that - we could relatively safely assume all species will be in database already (and therefor have an ID).

Reply:
I agree. I have changed the object to immutable. I have also dropped all the DAO interface methods except for getPokemonSpeciesById and getAllPokemonSpecies. As we do not have any species maintenance it might be better not having them in the database but rather in a static XML file.